chainer/link.py

```diff
@@ -211,9 +272,27 @@ def add_persistent(self, name, value):
             raise AttributeError(
                 'cannot register a new persistent value %s: attribute exists'
                 % name)
-        self._persistent.append(name)
+        self._persistent.add(name)
+        self._params.discard(name)
         d[name] = value
 
+    def register_persistent(self, name):
+        """Registers an attribute of a given name as a persistent value.
+
+        This is a convenient method to register an existing attribute as a
+        persistent value.
+
+        Args:
+            name (str): Name of the attribute to be registered.
+
+        """
```

Comment:
I found that this behavior is strange (basically we do not need to register a parameter as a persistent value). I will fix this point.